static/src/javascripts/projects/common/modules/navigation/search.js

```diff
@@ -0,0 +1,165 @@
+        let searchLoader;
+        const googleSearchSwitch = config.get('switches.googleSearch');
+        const googleSearchUrl = config.get('page.googleSearchUrl');
+        const googleSearchId = config.get('page.googleSearchId');
+
+        if (googleSearchSwitch && googleSearchUrl && googleSearchId && toggle) {
+            this.gcsUrl = `${googleSearchUrl}?cx=${googleSearchId}`;
+            this.resultSetSize =
+                config.get('page.section') === 'identity' ? 3 : 10;
+
+            searchLoader = throttle(() => {
+                this.load();
+            });
+
+            bean.on(toggle, 'click', e => {
+                const popup = document.getElementsByClassName(
+                    'js-search-popup'
+                )[0];
+                const maybeDismissSearchPopup = event => {
+                    let el = event.target;
+                    let clickedPop = false;
+
+                    if (popup) {
+                        while (el && !clickedPop) {
+                            /* either the search pop-up or the autocomplete resultSetSize
+                               NOTE: it would be better to check for `.gssb_c`,
+                                     which is the outer autocomplete element, but
+                                     google stops the event bubbling earlier
+                            */
+                            if (
+                                el &&
+                                el.classList &&
+                                (el.classList.contains('js-search-popup') ||
+                                    el.classList.contains('gsq_a'))
+                            ) {
+                                clickedPop = true;
+                            }
+
+                            el = el.parentNode;
+                        }
+
+                        if (!clickedPop) {
+                            event.preventDefault();
+                            toggle.classList.remove('is-active');
+                            popup.classList.add('is-off');
+                            bean.off(
+                                document,
+                                'click',
+                                maybeDismissSearchPopup
+                            );
+                        }
+                    }
+                };
+
+                setTimeout(() => {
+                    if (toggle.classList.contains('is-active')) {
+                        bean.on(document, 'click', maybeDismissSearchPopup);
+                    }
+                });
+
+                searchLoader();
+                // Make sure search is always in the correct state
+                focusSearchField();
+                e.preventDefault();
+            });
+        }
+    }
+
+    load(): void {
+        let s;
+        let x;
+
+        const containers = [
+            ...document.querySelectorAll('.js-search-placeholder'),
```

Comment:
Can we change this to  `...document.getElementsByClassName('js-search-placeholder')`?